src/web/history.rkt

```diff
@@ -147,3 +147,107 @@
                         "\\[\\leadsto "
                         ,(core->tex step-prog #:loc (cons 2 loc) #:color "blue")
                         "\\]"))))))))
+
+
+(define (render-json altn pcontext pcontext2 ctx)
+  (define repr (context-repr ctx))
+  (define err
+    (format-accuracy (errors-score (errors (alt-expr altn) pcontext ctx)) (representation-total-bits repr) #:unit "%"))
+  (define err2
+    (format-accuracy (errors-score (errors (alt-expr altn) pcontext2 ctx)) (representation-total-bits repr) #:unit "%"))
+
+  (match altn
+    [(alt prog 'start (list))
+     `#hash((program . ,(fpcore->string (program->fpcore prog ctx))) (type . "start") (error . ,err) (training-error . ,err2))]
+
+    [(alt prog `(start ,strategy) `(,prev))
+     `#hash(
+             (program . ,(fpcore->string (program->fpcore prog ctx))) 
+             (type . "strategy") 
+             (strategy . ,(~a strategy)) 
+             (prev . ,(render-json prev pcontext pcontext2 ctx))
+             (error . ,err) 
+             (training-error . ,err2))]
+
+    [(alt prog `(regimes ,splitpoints) prevs)
+     (define intervals
+       (for/list ([start-sp (cons (sp -1 -1 #f) splitpoints)] [end-sp splitpoints])
+         (interval (sp-cidx end-sp) (sp-point start-sp) (sp-point end-sp) (sp-bexpr end-sp))))
+
+     `#hash(
+             (program . ,(fpcore->string (program->fpcore prog ctx))) 
+             (type . "regimes") 
+             (conditions . ,(for/list ([entry prevs] [idx (in-naturals)]) 
+                (let ([entry-ivals (filter (λ (intrvl) (= (interval-alt-idx intrvl) idx)) intervals)]) (map (curryr interval->string repr) entry-ivals))))
+             (prev . ,(for/list ([entry prevs] 
```

Comment:
Maybe this should be called `prevs`, so that we don't have different types having `prev` fields with different types.